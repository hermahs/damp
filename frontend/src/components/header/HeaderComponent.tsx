import * as React from 'react';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import { Card, Modal, CardContent, Button } from '@mui/material';
import { useState } from 'react';
import ArrowBackIcon from '@mui/icons-material/ArrowBack';

//import Steam from '../../images/steam.png';


const style = {
  position: 'absolute' as 'absolute',
  top: '50%',
  left: '50%',
  transform: 'translate(-50%, -50%)',
  width: '60%',
  bgcolor: 'background.paper',
  border: '2px solid #000',
  boxShadow: 24,
  p: 4,
  overflow: "hidden",
  overflowY: "scroll",
};
const styleCard = {
  position: 'absolute' ,
  top: '50%',
  left: '50%',
  transform: 'translate(-50%, -50%)',
  width: '80%',
  height: 500,
  overflow: "hidden",
  overflowY: "scroll",
};

export function Header() {
  const [openModalInfo, setOpenModalInfo] = useState(false);
  const [darkMode, setDarkMode] = useState(false);

  const handleOpen = () => setOpenModalInfo(true);
  const handleClose = () => setOpenModalInfo(false);
  // const handleColor = () => 
  //{
  //   (theme === "theme") ? setTheme("darkTheme") : setTheme("theme");    
  // };

  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar position='relative'>
        <Toolbar>
          {/** <img src={Steam} alt="Steam" style={{ width: 30, height: 30, borderRadius: 50, marginRight: 6 }}  /> */}
          <Typography variant="h5" sx={{ flexGrow: 1 }}>
            DAMP
          </Typography>
          <Button color="inherit" onClick={() => setDarkMode(!darkMode)}>
            DarkMode
          </Button>
          <Button color="inherit" onClick={handleOpen}>
            ABOUT
          </Button>
        </Toolbar>
      </AppBar>
      <Modal
        open={openModalInfo}
        onClose={handleClose}
        aria-labelledby="modal-modal-title"
        aria-describedby="modal-modal-description"
        sx={{padding: 4}}>
        <Card sx={{...styleCard}} style={{maxWidth: "650px"}}>
          <CardContent>
              <ArrowBackIcon onClick={handleClose}/>
              <Typography id="modal-modal-title" variant="h5" component="h2" align='center'>
                What is Damp?
              </Typography>
              <Typography id="modal-modal-description" sx={{ mt: 2 }} color="text.primary">
                Damp is a library of games delivered by the popular video game digital distribution service Steam. 
                Today the platform is the largest digital distribution platform for PC games, with over 100 million active users and more than 50 000 games.
                Damp fetches data from the Steam API and displays it in a user-friendly way.
                The user can search for games, see the most important details about them, and write comments to share with other users.
              </Typography>
              <Typography id="modal-modal-title" variant="h5" component="h2" sx={{ mt: 4 }} align='center'>
                How to use Damp?
              </Typography>
              <Typography id="modal-modal-description" sx={{ mt: 2 }} color="text.primary" >
                To get started, you can search for a game by name, developer or publisher.
                In the search results, you can see the game's name, release date, developer, publisher and a brief description.
                You can also see the game's cover image and write a comment about the game.
                In addition, you are able to filter the games on several categories and sort the results of your choice.
              </Typography>
          </CardContent>        
        </Card>
      </Modal>
    </Box>
  );
}
