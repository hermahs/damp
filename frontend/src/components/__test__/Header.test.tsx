import { Header } from "../header";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import { useEffect, useState } from "react";
import { act } from "react-dom/test-utils";

function TestComponent() {
  const [darkMode, setDarkMode] = useState(false);

  useEffect(() => {
    const val = localStorage.getItem("dark-mode");
    if (val === "true") {
      setDarkMode(true);
    }
  }, [darkMode]);

  return (
    <>
      <p>Is it on? {JSON.stringify(darkMode)}</p>
      <Header darkMode={darkMode} setDarkMode={setDarkMode} />
    </>
  );
}
describe("test header component", () => {
  it("renders with correct header", () => {
    render(<Header darkMode={false} setDarkMode={() => {}} />);
    screen.getByText(/damp/i);
    screen.getByText(/darkmode/i);
    screen.getByText(/about/i);
  });

  it("about loads correctly", async () => {
    render(<Header darkMode={false} setDarkMode={() => {}} />);
    screen.getByText(/about/i).click();
    act(() => {
      screen.getByText(/darkmode/i).click();
    });
    const cont = await waitFor(() => screen.findByText(/how to use damp/i));

    fireEvent.keyDown(cont, {
      key: "Escape",
      code: "Escape",
      keyCode: 27,
      charCode: 27,
    });

    const modal = screen.queryByText(/how to use damp/i)
    expect(modal).not.toBeInTheDocument()
  });

  it("handles darkmode toggle", async () => {
    localStorage.setItem("dark-mode", "false");
    render(<TestComponent />);
    screen.getByText(/false/i);
    act(() => {
      screen.getByText(/darkmode/i).click();
    });
    await waitFor(() => screen.findByText(/true/i));
    act(() => {
      screen.getByText(/darkmode/i).click();
    });
    await waitFor(() => screen.findByText(/false/i));
  });
});
