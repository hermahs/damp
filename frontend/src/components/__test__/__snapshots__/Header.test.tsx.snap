// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header snapshot tests basic snapshot test 1`] = `
<div
  className="MuiBox-root css-i9gxme"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionRelative css-ep400p-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 css-1f8a19r-MuiTypography-root"
        onClick={[Function]}
      >
        DAMP
      </h5>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Light
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        ABOUT
      </button>
    </div>
  </header>
</div>
`;
